src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -282,8 +357,8 @@ public void setIntervalsForTraversal( final List<SimpleInterval> intervals ) {
      * @return a List of all Features in this data source that overlap the provided interval
      */
     public List<T> queryAndPrefetch( final SimpleInterval interval ) {
-        if ( ! hasIndex ) {
-            throw new UserException("File " + featureFile.getAbsolutePath() + " requires an index to enable queries by interval. " +
+        if ( ! supportsRandomAccess ) {
+            throw new UserException("File " + featureInput.getFeaturePath() + " requires an index to enable queries by interval. " +
                                     "Please index this file using the bundled tool " + IndexFeatureFile.class.getSimpleName());
```

Comment:
Generalize this error message: "Input x must support random access to enable queries by interval. If it's a file, please index it using the bundled tool IndexFeatureFile"